### ISSUE TITLE
Rename component causes semantic colors be disappear for that component until edit

### ISSUE THREAD

**Issue by ryanbrandenburg** [dotnet/aspnetcore#25045]
1. Create new Blazor app
1. Open Index.razor and rename SurveyPrompt to SurveyPrompt2.
1. See that the color changes and never stabalizes back to what it was.
1. Make an edit and it will return to the old color.

This is probably because the semantic edit request is made before the new component is recognized by project.razor.json.

**Comment by NTaylorMullen** [comment]
We think this has been fixed but should verify

**Comment by NTaylorMullen** [comment]
So looks like this still happens IF SurveyPrompt is open in the editor. Investigating.

**Comment by NTaylorMullen** [comment]
Found the root cause to be in the LSP platform: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1215725

I'll see if I can contribute a fix to them to unblock us.

**Comment by NTaylorMullen** [comment]
Was able to fix the root cause but exposed another underlying problem in semantic tokens. @gundermanc is coming to the rescue to resolve that on the LSP platform front. That being said it looks like there was another issue uncovered as part of this. Our ITextBufferCOntentTypeChangeListener approach isn't triggered when files are renamed while they're opened. Reaching out to CoreEditor to see if this is expected.

**Comment by gundermanc** [comment]
>Our ITextBufferCOntentTypeChangeListener approach isn't triggered when files are renamed while they're opened. 

Are you changing just the name portion or the file extension too? This even only fires if you change the content type, say, by changing the file extension.

**Comment by NTaylorMullen** [comment]
> Are you changing just the name portion or the file extension too?

Just the name, aka `Component1.razor` => `ComponentRenamed.razor`

**Comment by gundermanc** [comment]
That's expected then. Content type should only change if you change the file extension. If you want to listen for an underlying document rename, you will probably need to use the ITextDocumentFactoryService to look up the ITextDocument for the ITextBuffer. It exposes a file renamed event.

Is the incorrect code in the LSP client, and if so, is there a bug tracking this in the Editor or Extensibility area path?

**Comment by NTaylorMullen** [comment]
> That's expected then. Content type should only change if you change the file extension. If you want to listen for an underlying document rename, you will probably need to use the ITextDocumentFactoryService to look up the ITextDocument for the ITextBuffer. It exposes a file renamed event.

Ya I ended up finding out the same after some digging. Looks like the LSP client already does the right thing for this and listens for FileActionOccurred events off of the ITextDocument's. The hard part is that we moved over to the ITextBufferContentTypeChangedListener tech with the intent to not have to subscribe to several different mechanisms to understand buffer state. Sounds like we still have to do some level of that sadly though 😢

**Comment by NTaylorMullen** [comment]
Note: Rename component does not work in Codespaces

**Comment by ryanbrandenburg** [comment]
We're still seeing this in VS.

**Comment by ryanbrandenburg** [comment]
Context https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1247479.
